app/models/user.rb

```diff
@@ -1,2 +1,7 @@
 class User < ApplicationRecord
+  has_many :posts, foreign_key: :user_id
+  has_many :wall_posts, foreign_key: :receiver_id
```

Comment:
How does this association work?

Reply:
This is to get all the posts on someones "wall".

Reply:
whats the difference with `posts` and `wall_posts`?

Reply:
Posts are posts that the user made
Wall Posts are posts on the users wall (could be renamed to timeline)
```ruby
has_many :timeline, foreign_key: :receiver_id, class_name: 'Post'
```